raiden/transfer/mediated_transfer/initiator.py

```diff
@@ -0,0 +1,292 @@
+# -*- coding: utf-8 -*-
+from copy import deepcopy
+
+from raiden.transfer.architecture import Iteration
+from raiden.transfer.state import RoutesState
+from raiden.transfer.mediated_transfer.state import (
+    InitiatorState,
+    LockedTransferState,
+)
+from raiden.transfer.mediated_transfer.transition import update_route
+from raiden.transfer.state_change import (
+    ActionCancelTransfer,
+    ActionRouteChange,
+    Block,
+)
+from raiden.transfer.mediated_transfer.state_change import (
+    ActionCancelRoute,
+    ActionInitInitiator,
+    ReceiveSecretRequest,
+    ReceiveSecretReveal,
+    ReceiveTransferRefund,
+)
+from raiden.transfer.mediated_transfer.events import (
+    EventTransferFailed,
+    SendMediatedTransfer,
+    SendRevealSecret,
+)
+from raiden.utils import sha3
+
+
+def cancel_current_route(state):
+    """ Clear current state and try a new rotue.
+
+    - Discards the current secret
+    - Add the current route to the canceled list
+    - Add the current message to the canceled transfers
+    """
+    assert state.revealsecret is None, 'cannot cancel a transfer with a RevealSecret in flight'
+
+    state.routes.canceled_routes.append(state.route)
+    state.canceled_transfers.append(state.message)
+
+    state.transfer.secret = None
+    state.transfer.hashlock = None
+    state.message = None
+    state.route = None
+    state.secretrequest = None
+
+    return try_new_route(state)
+
+
+def user_cancel_transfer(state):
+    """ Cancel the current in-transit message. """
+    assert state.revealsecret is None, 'cannot cancel a transfer with a RevealSecret in flight'
+
+    state.transfer.secret = None
+    state.transfer.hashlock = None
+    state.message = None
+    state.route = None
+    state.secretrequest = None
+    state.revealsecret = None
+
+    cancel = EventTransferFailed(
+        identifier=state.transfer.identifier,
+        reason='user canceled transfer',
+    )
+    iteration = Iteration(None, [cancel])
+
+    return iteration
+
+
+def try_new_route(state):
+    assert state.route is None, 'cannot try a new route while one is being used'
+
+    # TODO:
+    # - Route ranking. An upper layer should rate each route to optimize
+    #   the fee price/quality of each route and add a rate from in the range
+    #   [0.0,1.0].
+    # - Add in the policy a per route:
```

Comment:
Yes I undestand but still the current comment `Add in the policy a per route:` sounds wrong from a grammatic pespective.